src/Simplex/Chat/Controller.hs

```diff
@@ -193,6 +196,28 @@ data ChatController = ChatController
     logFilePath :: Maybe FilePath
   }
 
+data RemoteHostSession = RemoteHostSession
+  { -- | A process that relays the commands to its host
+    handler :: Async (),
+    -- | Path for local resources to be synchronized with host
+    path :: FilePath,
+    -- | Commands to be relayed to a host
+    command :: forall m . ChatMonad m => ByteString -> m ChatResponse,
```

Comment:
I don't think I'm a fan of this pattern when methods are captured as part of the state - why not just to pass session to the function that does it? This pattern feels opposite of functional, almost like doing Java in Haskell.